### ISSUE TITLE
BUG: scipy.interpolate.BSpline gives wrong result in version 1.8.1-1

### ISSUE THREAD

**Issue by Shengyis** [scipy/scipy#16448]
### Describe your issue.

Following the example code scipy document,  just change the evaluation point 
```
from scipy.interpolate import BSpline

k = 2
t = [0, 1, 2, 3, 4, 5, 6]
c = [-1, 2, 0, -1]
spl = BSpline(t, c, k)

def B(x, k, i, t):
   if k == 0:
      return 1.0 if t[i] <= x < t[i+1] else 0.0
   if t[i+k] == t[i]:
      c1 = 0.0
   else:
      c1 = (x - t[i])/(t[i+k] - t[i]) * B(x, k-1, i, t)
   if t[i+k+1] == t[i+1]:
      c2 = 0.0
   else:
      c2 = (t[i+k+1] - x)/(t[i+k+1] - t[i+1]) * B(x, k-1, i+1, t)
   return c1 + c2

def bspline(x, t, c, k):
   n = len(t) - k - 1
   assert (n >= k+1) and (len(c) >= n)
   return sum(c[i] * B(x, k, i, t) for i in range(n))

print(spl(1.25))
print(bspline(1.25, t, c, k))
```
the two results should be the same but actually, they are different. 

-3.15625
-0.625

the second one -0.625 is correct. 
If you change the evaluation point to 2.5 which is the one used in the document, then the results are the same. 
The main reason is that it seems for order k, the returned B_{i, k}(x) with i <= k or i > n is not correct. For example, If I change coefficient c= [-1, 2, 0, -1] to c = [1, 0, 0, 0] and plot spl(x), this should give me B_{0, 1}(x), which is like hat but actually it only gives right half hat. Same that B_{1, 1}(x) is also wrong. 

### Reproducing Code Example

```python
from scipy.interpolate import BSpline

k = 2
t = [0, 1, 2, 3, 4, 5, 6]
c = [-1, 2, 0, -1]
spl = BSpline(t, c, k)

def B(x, k, i, t):
   if k == 0:
      return 1.0 if t[i] <= x < t[i+1] else 0.0
   if t[i+k] == t[i]:
      c1 = 0.0
   else:
      c1 = (x - t[i])/(t[i+k] - t[i]) * B(x, k-1, i, t)
   if t[i+k+1] == t[i+1]:
      c2 = 0.0
   else:
      c2 = (t[i+k+1] - x)/(t[i+k+1] - t[i+1]) * B(x, k-1, i+1, t)
   return c1 + c2

def bspline(x, t, c, k):
   n = len(t) - k - 1
   assert (n >= k+1) and (len(c) >= n)
   return sum(c[i] * B(x, k, i, t) for i in range(n))

print(spl(1.25))
print(bspline(1.25, t, c, k))
```

### Error message

```shell
-3.15625
-0.625
```

### SciPy/NumPy/Python version information

1.8.1 1.22.4 sys.version_info(major=3, minor=10, micro=5, releaselevel='final', serial=0)sys.version_info)

**Comment by ev-br** [comment]
tl;dr: Not a bug. The equivalence only holds for `t[k] < x < t[-k]`, which in this case is 

```
>>> t[k:-k]
[2, 3, 4]
```
and the OP evaluates it at `x = 1.25`.

The OP code snippet is from https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.BSpline.html

The text literally just above the code sample reads _Here we construct a quadratic spline function on the base interval 2 <= x <= 4 ..._ 

and just below it, _Note that outside of the base interval results differ. This is because [BSpline](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.BSpline.html#scipy.interpolate.BSpline) extrapolates the first and last polynomial pieces of B-spline functions active on the base interval._

So, to conclude:
- there is no bug
- If you need individual basis elements, including those at the edges of the interval, use `BSpline.basis_element`
- this behavior may indeed be confusing 
- if you have an idea of how to improve the documentation: the wording of the BSpline class docstring or elsewhere --- please send a PR, we will gladly review.

**Comment by ev-br** [comment]
I'm closing this ticket but feel free to discuss further. And since you clearly invested time into this matter, you are a perfect person to suggest improvements to our docs --- please do!

**Comment by Shengyis** [comment]
many thanks! I should read the document more carefully.
